src/Symfony/Component/Validator/Tests/Mapping/ClassMetadataTest.php

```diff
@@ -308,6 +309,19 @@ public function testGetPropertyMetadataReturnsEmptyArrayWithoutConfiguredMetadat
     {
         $this->assertCount(0, $this->metadata->getPropertyMetadata('foo'), '->getPropertyMetadata() returns an empty collection if no metadata is configured for the given property');
     }
+
+    public function testClassMetadataMembersAreUniqueByPropertyAndByClassAfterMergingConstraints()
+    {
+        $metadata = new ClassMetadata(self::INTERFACE_A_CLASS);
+        $metadata->addGetterConstraint('data', new ConstraintA());
+
+        $otherMetadata = new ClassMetadata(self::INTERFACE_A_CLASS);
+        $otherMetadata->addGetterConstraint('data', new ConstraintB());
+
+        $metadata->mergeConstraints($otherMetadata);
+
+        $this->assertCount(1, $metadata->getPropertyMetadata('data'));
```

Comment:
Not sure I would agree that they are always "the same" in "real-life" :wink: Seems like a BC break to me.

Imho we would have to check if the constraints are actually 100% the same, so same class & same options. Only then we can have real duplicates.